### DISCUSSION TITLE
LM-Infinite: Simple On-the-Fly Length Generalization for LLM

### DISCUSSION THREAD

**Discussion by Jipok** [ggml-org/llama.cpp#2936] [Ideas]
https://www.reddit.com/r/LocalLLaMA/comments/165zpn9/r_lminfinite_simple_onthefly_length/
https://arxiv.org/pdf/2308.16137.pdf

![изображение](https://github.com/ggerganov/llama.cpp/assets/25588359/639199d3-46c7-4e6b-bd0a-9b67b387c7e3)

Any chance this will be implemented in llama.cpp? As far as I understand, nothing in theory prevents this from being done.

**Comment by kyegomez** [comment]
Hey, I have a very basic implementation here
https://github.com/kyegomez/LM-Infinite

**Comment by Glaciohound** [comment]
Hello, we have a more detailed implementation here:

https://github.com/Glaciohound/LM-Infinite.

Probably you will be interested in checking out.
